### PR TITLE
[SYCL][ESIMD][EMU] Enable dpas with ESIMD_EMULATOR backend

### PR DESCRIPTION
**Pull request by dongkyunahn-intel**
- +fix for 'saturate<>' for ESIMD_EMULATOR backend

### CODE REVIEW

**Comment by dongkyunahn-intel** [comment]
Note : DPASW support will be enabled later as it requires updates in esimd_emulator backend interface as well as CM_EMU library for runtime support

**Review by v-klochkov — DISMISSED** [review]

**Comment by pvchupin** [comment]
Thanks @dongkyunahn-intel, do you know anything about SYCL :: unused_spec_const.cpp ?

**Comment by dongkyunahn-intel** [comment]
> Thanks @dongkyunahn-intel, do you know anything about SYCL :: unused_spec_const.cpp ?

@pvchupin , as far as I know 'spec_const' tests are disabled for ESIMD_EMULATOR backend as they are failing with `Unsupported 'online_compiler'` error. It could be because of 'AOT' that is not supported by the ESIMD_EMULATOR backend. Currently, there is no plan for enabling those tests with the backend.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 1]
```diff
@@ -110,6 +110,8 @@ __ESIMD_INTRIN __ESIMD_raw_vec_t(T, N)
 }
 #endif // __SYCL_DEVICE_ONLY__
 
+#ifdef __SYCL_DEVICE_ONLY__
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:113]: The agreed way to structure intrinsics is
```
void __esimd_intrinsic(...)
#ifdef __SYCL_DEVICE_ONLY__
;
#else
{ <emu_implemenation_of_intrinsic>; }
#endif // __SYCL_DEVICE_ONLY__
```
this way we guarantee same signatures on host and device.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 2]
```diff
@@ -457,12 +459,13 @@ __esimd_dpas_bits_precision(__ESIMD_ENS::argument_type precisionType) {
...
 }
 
+// TODO : 'src0' argument type other than 'void*'?
 template <__ESIMD_ENS::argument_type src1_precision,
           __ESIMD_ENS::argument_type src2_precision, int systolic_depth,
           int repeat_count, typename RT, typename T1, typename T2,
           __ESIMD_NS::uint SZ, __ESIMD_NS::uint N1, __ESIMD_NS::uint N2>
 inline __ESIMD_DNS::vector_type_t<RT, SZ>
-__esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
+__esimd_dpas_inner(const void *src0,
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:468]: why src0 needs to be void* ?

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 1 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:113]: I don't know why, but `__esimd_dpas*` function signatures are different per __SYCL_DEVICE_ONLY__ from caller side. For example, `dpas_info` argument does not exist for without __SYCL_DEVICE_ONLY__.

https://github.com/intel/llvm/blob/dd17a3363e2538391cfd74b3dbdaa39b13827202/sycl/include/sycl/ext/intel/experimental/esimd/math.hpp#L2025

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 2 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:468]: For following tests, `__ESIMD_DNS::vector_type_t<RT, SZ> *` is working fine.

```
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/dpas/dpas_test1.cpp
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/dpas/dpas_test2.cpp
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/dpas/dpas_test3.cpp
```

However, following test fails.

```
https://github.com/intel/llvm/blob/sycl/sycl/test/esimd/dpas.cpp
```

Compilation error message : 
```
llvm_build/install/bin/../include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:669:29: error: static_cast from 'short * __attribute__((ext_vector_type(16)))' to 'sycl::ext::intel::esimd::detail::vector_type_t<floa
t, 16> *' (aka 'cl::sycl::ext::intel::esimd::detail::raw_vector_type<float, 16>::type *') is not allowed
                            static_cast<__ESIMD_DNS::vector_type_t<T, N> *>(std::addressof(src0)),
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

As I was not sure what type of argument could work, I used `void *` (with TODO comment)

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 1 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: `__esimd_dpas*` functions are changed to have same function signature regardless of `__SYCL_DEVICE_ONLY__` and their definitions are changed as suggested.

Previously, `src1_precision`, `src2_precision`, `systolic_depth`, and `repeat_count` are template arguments for emulation while they are runtime arguments for (actual) devices. I converted all of those template arguments into runtime arguments so that function signatures are same between with and without `__SYCL_DEVICE_ONLY__`.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 2 above)
Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:441]: I think this needs to be resolved before it can be merged

Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:441]: host and device signatures should always match exactly

Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:441]: They do for `__esimd_dpas*` intrinsics used by `dpas/w` from `math.hpp'. This `__esimd_dpas_inner` is only for host/esimd_emulator.

Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:441]: I see, thanks. Still, void* should be avoided as it decreases readability and can be source of bugs.
Also, please roll-back unrelated formatting changes.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 3]
```diff
@@ -513,54 +492,55 @@ __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
   constexpr size_t SIMDSize = 8;
 #endif
 
-  constexpr bool
-      pvcHfDest = isPvc && std::is_same<RT, __ESIMD_EMU_DNS::half>::value,
-      pvcBfDest = isPvc && std::is_same<RT, short>::value,
-      pvcBfOrHfDest = pvcBfDest || pvcHfDest,
+  constexpr bool pvcHfDest =
+                     isPvc && std::is_same<RT, __ESIMD_EMU_DNS::half>::value,
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:496]: please rollback formatting chanhes

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 4]
```diff
@@ -513,54 +492,55 @@ __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
...
+                 pvcBfDest = isPvc && std::is_same<RT, short>::value,
+                 pvcBfOrHfDest = pvcBfDest || pvcHfDest;
 
-      pvcBfDestChecks = pvcBfDest &&
-                        src1_precision == __ESIMD_ENS::argument_type::BF16 &&
-                        src2_precision == __ESIMD_ENS::argument_type::BF16,
+  bool pvcBfDestChecks = pvcBfDest &&
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:500]: why is this no longer constexpr?

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 5]
```diff
@@ -513,54 +492,55 @@ __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
...
+                         src1_precision == __ESIMD_ENS::argument_type::BF16 &&
+                         src2_precision == __ESIMD_ENS::argument_type::BF16,
 
-      pvcHfDestChecks =
-          pvcHfDest && ((src1_precision == __ESIMD_ENS::argument_type::FP16 &&
-                         src2_precision == __ESIMD_ENS::argument_type::FP16) ||
-                        (src1_precision == __ESIMD_ENS::argument_type::BF16 &&
-                         src2_precision == __ESIMD_ENS::argument_type::BF16)),
+       pvcHfDestChecks =
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:504]: here and in other places - please avoid unrelated formatting  changes.

Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:504]: I see. The formatting was done by `clang-format`. I'll revert this part and guard with `// clang-format off`.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 4 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:500]: As `src1_precision` and `src2_precision` are changed to runtime argument (from template argument).

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 3 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Done

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 5 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Done.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 2 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: I reverted this `void*` work-around as the failing test is passing locally.

**Comment by dongkyunahn-intel** [comment]
Unrelated failure : 
```
SYCL / Linux / CUDA LLVM Test Suite
```

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 2 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: `void*` argument is no longer used and tests are passing (with unrelated failures from 'SYCL / Linux / CUDA LLVM Test Suite')

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 6]
```diff
@@ -12,6 +12,8 @@
 
 /// @cond ESIMD_DETAIL
 
+#define ARG_UNUSED(x) (void)x
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:15]: Please
1) move this to some utility header
2) use __ESIMD_ prefix to avoid polluting user namespace
3) consider adding #ifndef __SYCL_DEVICE_ONLY__

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 7]
```diff
@@ -439,7 +416,7 @@ __ESIMD_INTRIN __ESIMD_raw_vec_t(T, SZ)
   return retv;
 }
 
-inline constexpr __ESIMD_NS::uint
+inline __ESIMD_NS::uint
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:419]: why this change? this may lead to inefficiency or inability to use it in template meta-programming.

Note: you may change this to `if constexpr(...)` style to avoid evaluation of `false` branches (and potential related errors)

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 8]
```diff
@@ -457,14 +434,15 @@ __esimd_dpas_bits_precision(__ESIMD_ENS::argument_type precisionType) {
...
-          int repeat_count, typename RT, typename T1, typename T2,
-          __ESIMD_NS::uint SZ, __ESIMD_NS::uint N1, __ESIMD_NS::uint N2>
+template <typename RT, typename T1, typename T2, __ESIMD_NS::uint SZ,
+          __ESIMD_NS::uint N1, __ESIMD_NS::uint N2>
 inline __ESIMD_DNS::vector_type_t<RT, SZ>
 __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
                    const __ESIMD_DNS::vector_type_t<T1, N1> &src1,
-                   const __ESIMD_DNS::vector_type_t<T2, N2> &src2) {
+                   const __ESIMD_DNS::vector_type_t<T2, N2> &src2,
+                   __ESIMD_ENS::argument_type src1_precision,
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:443]: this should probably return to compile-time evaluation

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 9]
```diff
@@ -473,7 +451,7 @@ __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
       __ESIMD_EMU_DNS::SetSatur<T2,
                                 __ESIMD_EMU_DNS::is_inttype<RT>::value>::set();
 
-  constexpr __ESIMD_NS::uint ops_per_chan =
+  __ESIMD_NS::uint ops_per_chan =
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:454]: ditto (here and below)

**Review by kbobrovs — CHANGES_REQUESTED** [review]

**Comment by dongkyunahn-intel** [comment]
Function signatures for `__esimd_dpas*` will be changed to have `precision`, `systolic depth`, and `repeat count` (from `dpas_info`) as template arguments. Host implementation will be changed in this PR. Device implementation will be changed later.

File: sycl/include/sycl/ext/intel/experimental/esimd/common.hpp [hunk 10]
```diff
@@ -410,6 +410,39 @@ constexpr void check_lsc_cache_hint() {
   }
 }
 
+ESIMD_INLINE
```
[the remark this answers is not in the document]
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/common.hpp]: Change is removed.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 6 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: `ARG_USED` is removed as it is no longer used.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 8 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Change is removed.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 9 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Change is removed.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 7 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: This change is removed.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 11]
```diff
@@ -110,6 +110,11 @@ __ESIMD_INTRIN __ESIMD_raw_vec_t(T, N)
 }
 #endif // __SYCL_DEVICE_ONLY__
 
+#ifdef __SYCL_DEVICE_ONLY__
+
+// TODO: For Device compilation, change 'dpas_info' function argument
+// (precision, depth, repeat, signedness) into template argument like
+// __esimd_dpas* intrinsic declarations for host mode
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:117]: I suggest to remove duplicates of this TODO and change it as follows:
```
// TODO: __esimd_dpas* should have single declaration for host and device:
// void __esimd_dpas*(...)
// #ifdef __SYCL_DEVICE_ONLY__
//  ;
// #else
// { <emu_implemenation_of_intrinsic>; }
// #endif // __SYCL_DEVICE_ONLY__
```

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 12]
```diff
@@ -647,7 +652,8 @@ __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
             static_cast<short>(reinterpret_cast<uint32_t &>(tmpUint) >> 16);
       } else
         retv[r * SIMDSize + n] =
-            __ESIMD_EMU_DNS::satur<RT>::saturate(simdAcc[n], sat1);
+            __ESIMD_EMU_DNS::satur<RT>::template saturate<TmpAccEl>(simdAcc[n],
+                                                                    (int)sat1);
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:656]: why this cast?

Comment (dongkyunahn-intel) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:656]: I'll remove `(int)` cast.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 13]
```diff
@@ -658,19 +664,14 @@ __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
...
              __ESIMD_DNS::vector_type_t<T2, N2> src2) {
-#ifdef __SYCL_EXPLICIT_SIMD_PLUGIN__
   return __esimd_dpas_inner<src1_precision, src2_precision, systolic_depth,
                             repeat_count, T, T1, T2, N, N1, N2>(
-      std::addressof(src0), src1, src2);
-#else  // __SYCL_EXPLICIT_SIMD_PLUGIN__
-  __ESIMD_UNSUPPORTED_ON_HOST;
-  return __ESIMD_DNS::vector_type_t<T, N>();
-#endif // __SYCL_EXPLICIT_SIMD_PLUGIN__
+      (__ESIMD_DNS::vector_type_t<T, N> *)std::addressof(src0), src1, src2);
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:674]: This cast seems very suspicious

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 2 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Type casting work-around is removed. Now `dpas.cpp` test fails with following error message.

```
/export/users/dongkyun/github/opensource/intel_llvm/build/bin/../include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:473:1: note: candidate function template not viable: no known conversion from 'short *__attribute__((ext_vector_type(16)))' to 'const sycl::ext::intel::esimd::detail::vector_type_t<float, 16U> *' (aka 'const cl::sycl::ext::intel::esimd::detail::raw_vector_type<float, 16>::type *') for 1st argument
__esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,

```

Command (From Check-in tests)

```
clang --driver-mode=g++  -D_GLIBCXX_ASSERTIONS=1 -DESIMD_XE_HPC -O0 -fsycl -c -Xclang -emit-llvm  $LLVM_SRC/sycl/test/esimd/dpas.cpp -o dpas.cpp.tmp
```

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 13 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: This change is removed. `dpas.cpp` now fails as follows:

```
/export/users/dongkyun/github/opensource/intel_llvm/build/bin/../include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:473:1: note: candidate function template not viable: no known conversion from 'short * __attribute__((ext_vector_type(16)))' to 'const sycl::ext::intel::esimd::detail::vector_type_t<float, 16U> *' (aka 'const cl::sycl::ext::intel::esimd::detail::raw_vector_type<float, 16>::type *') for 1st argument
__esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
```

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 12 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Done

Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Removed.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp (hunk 11 above)
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Done.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 14]
```diff
@@ -658,19 +664,14 @@ __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
 template <__ESIMD_ENS::argument_type src1_precision,
           __ESIMD_ENS::argument_type src2_precision, int systolic_depth,
           int repeat_count, typename T, typename T0, typename T1, typename T2,
-          int N, int N1, int N2>
+          int N, int N1, int N2, int sign_res, int sign_acc>
```
[the remark this answers is not in the document]
Comment (dongkyunahn-intel) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp]: Reverted.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 15]
```diff
@@ -110,6 +114,10 @@ __ESIMD_INTRIN __ESIMD_raw_vec_t(T, N)
 }
 #endif // __SYCL_DEVICE_ONLY__
 
+#ifdef __SYCL_DEVICE_ONLY__
+
+// TODO: __esimd_dpas* should have single declaration for host and device:
+// void __esimd_dpas*(...)
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:120]: it is not void actually `Ret __esimd_dpas*(...)`

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp [hunk 16]
```diff
@@ -662,15 +671,13 @@ template <__ESIMD_ENS::argument_type src1_precision,
 inline __ESIMD_DNS::vector_type_t<T, N>
 __esimd_dpas(__ESIMD_DNS::vector_type_t<T0, N> src0,
              __ESIMD_DNS::vector_type_t<T1, N1> src1,
-             __ESIMD_DNS::vector_type_t<T2, N2> src2) {
-#ifdef __SYCL_EXPLICIT_SIMD_PLUGIN__
+             __ESIMD_DNS::vector_type_t<T2, N2> src2, int sign_res,
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:674]: please remove the two unused arguments

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 17]
```diff
@@ -1854,8 +1854,8 @@ dpas(__ESIMD_NS::simd<T0, N> src0, __ESIMD_NS::simd<T1, N1> src1,
 #else
   __ESIMD_NS::simd<T, N> result =
       __esimd_dpas<src1_precision, src2_precision, systolic_depth, repeat_count,
-                   T, T0, T1, T2, N, N1, N2>(src0.data(), src1.data(),
-                                             src2.data());
+                   T, T0, T1, T2, N, N1, N2>(
+          src0.data(), src1.data(), src2.data(), dst_signed, src0_signed);
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1858]: please remove this change as well

**Review by kbobrovs — APPROVED** [review]

**Comment by dongkyunahn-intel** [comment]
Unrelated failures
```
SYCL / Linux / HIP AMDGPU LLVM Test Suite 
SYCL / Linux / CUDA LLVM Test Suite
```

**Comment by kbobrovs** [comment]
> Unrelated failures

What are the list of failing tests per each configuration?

**Comment by dongkyunahn-intel** [comment]
> > Unrelated failures
> 
> What are the list of failing tests per each configuration?

Updated the comment with failing tests.
